File: backend/src/main/java/com/sajjad/academicportalbackend/model/config/AttendanceRecord.java
package com.sajjad.academicportalbackend.model.config;

import com.sajjad.academicportalbackend.model.super_classes.AuditableEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
@Getter
@Setter
@Entity
public class AttendanceRecord extends AuditableEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Getter
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "student_id")
    private StudentProfile student;

    @Getter
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "course_id")
    private Course course;

    private LocalDate date;
    private String status;  // You can define an enum for better representation (present, absent, etc.)

    // Constructors, getters, and setters

    public void setStudent(StudentProfile student) {
        this.student = student;
    }

    public void setCourse(Course course) {
        this.course = course;
    }
}

File: frontend/src/app/model/config/attendance-record.ts
import { AuditAbleModel } from "../super-model/audit-able-model";
import { StudentProfile } from "./student-profile"; // Import StudentProfile if needed
import { Course } from "./course"; // Import Course if needed

export interface AttendanceRecord extends AuditAbleModel {
    id?: number;
    student?: StudentProfile; // Assuming StudentProfile is a related model
    course?: Course; // Assuming Course is a related model
    date?: Date; // Assuming you want to store the date as a JavaScript Date object
    status?: string; // You can define an enum for better representation (present, absent, etc.)
}
